src/repo.rs

```diff
@@ -2051,6 +2052,54 @@ impl Repository {
         }
         Ok(ignored == 1)
     }
+
+    /// Initializes a rebase operation to rebase the changes in `from_branch`
+    /// relative to `upstream` onto '`to_branch`. To begin the rebase
+    /// process, call `next()` or `operations()` on the returned object.
+    pub fn rebase_init(&self,
+                       from_branch: Option<AnnotatedCommit>,
+                       upstream: Option<AnnotatedCommit>,
+                       to_branch: Option<AnnotatedCommit>,
+                       options: Option<&mut RebaseOptions>)
+                       -> Result<Rebase, Error> {
+        unsafe {
+            let mut ret = 0 as *mut raw::git_rebase;
+            /*let from_branch_raw: *const raw::git_annotated_commit;
+            if from_branch.is_ok() {
+                from_branch_raw
+            }*/
+            let from_branch_raw = from_branch.as_ref()
+                .map(|ref a| a.raw() as *const raw::git_annotated_commit);
+            let upstream_raw = upstream.as_ref()
+                .map(|ref a| a.raw() as *const raw::git_annotated_commit);
+            let to_branch_raw = to_branch.as_ref()
+                .map(|ref a| a.raw() as *const raw::git_annotated_commit);
+            let options_raw = options.map(|opts| opts.raw());
+            try_call!(raw::git_rebase_init(&mut ret,
+                                           self.raw(),
+                                           from_branch_raw,
+                                           upstream_raw,
+                                           to_branch_raw,
+                                           options_raw.as_ref().map(|opts| opts as *const _)));
+            drop(from_branch);
+            drop(upstream);
+            drop(to_branch);
+            drop(options_raw);
+            Ok(Binding::from_raw(ret))
+        }
+    }
+    /// Opens an existing rebase that was previously started by either an
+    /// invocation of `rebase_init` or by another client.
+    pub fn rebase_open(&self, options: Option<RebaseOptions>) -> Result<Rebase, Error> {
```

Comment:
I think it should be safe for this to be `Option<&mut RebaseOptions>`

Reply:
Okay.